apps/gateway/lib/gateway/controllers/curse_of_mirra/character_controller.ex

```diff
@@ -9,27 +9,52 @@ defmodule Gateway.Controllers.CurseOfMirra.CharacterController do
   def get_characters_config(conn, _params) do
     case GameBackend.Units.Characters.get_curse_characters() do
       [] -> {:error, :not_found}
-      characters -> send_resp(conn, 200, Jason.encode!(transform_to_map_from_ecto_struct(characters)))
+      characters -> send_resp(conn, 200, Jason.encode!(encode_characters(characters)))
     end
   end
 
-  defp transform_to_map_from_ecto_struct(ecto_struct) when is_struct(ecto_struct) do
+  defp ecto_struct_to_map(ecto_struct) when is_struct(ecto_struct) do
```

Comment:
Why not using the Jason decoder in the ecto schemas instead of parsing them here? 

you can include something like:
```
@derive {Jason.Encoder, only: [fields]}
schema "characters" do
``` 

and then you can use  `Jason.encode!(characters)` what do you think?

Reply:
Hmmm I would have to try if it works. We can create an issue to check it later. What do you think?